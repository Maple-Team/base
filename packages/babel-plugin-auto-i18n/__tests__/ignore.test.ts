import { getTransformCode } from './test-help'

it('handle JSXText with comments', () => {
  const sourceCode = `
  export const Component = () => {
    return <div>{/*@i18n-ignore*/'你好'}<span>{/*@i18n-ignore*/}你好</span></div>
  }
`
  const result = getTransformCode(sourceCode, 'filename.tsx')
  expect(result?.code).toMatchSnapshot()
})

it('handle stringLiteral with comments', () => {
  const sourceCode = `
  export const Component = () => {
    const var1 = /*@i18n-ignore*/'你好'
    return <div>你好</div>
  }
`
  const result = getTransformCode(sourceCode, 'filename.tsx')
  expect(result?.code).toMatchSnapshot()
})

it('handle templateElement with comments', () => {
  const sourceCode = `
  export const Component = () => {
    const var1 = /*@i18n-ignore*/\`templateElement你好\`;
    return <div>你好</div>
  }
`
  const result = getTransformCode(sourceCode, 'filename.tsx')
  expect(result?.code).toMatchSnapshot()
})
