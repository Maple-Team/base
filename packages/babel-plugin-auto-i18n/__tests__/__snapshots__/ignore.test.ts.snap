// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`handle JSXText with comments 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.Component = void 0;
var _reactI18next = require(\\"react-i18next\\");
var Component = exports.Component = function Component() {
  var _useTranslation = (0, _reactI18next.useTranslation)(),
    t = _useTranslation.t;
  return <div>{'你好'}<span data-i18n=\\"[\\\\\\"key_\\\\u4F60\\\\u597D\\\\\\"]\\">{/*@i18n-ignore*/}{t(\\"key_你好\\")}</span></div>;
};"
`;

exports[`handle stringLiteral with comments 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.Component = void 0;
var _reactI18next = require(\\"react-i18next\\");
var Component = exports.Component = function Component() {
  var _useTranslation = (0, _reactI18next.useTranslation)(),
    t = _useTranslation.t;
  var var1 = '你好';
  return <div data-i18n=\\"[\\\\\\"key_\\\\u4F60\\\\u597D\\\\\\"]\\">{t(\\"key_你好\\")}</div>;
};"
`;

exports[`handle templateElement with comments 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.Component = void 0;
var _reactI18next = require(\\"react-i18next\\");
var Component = exports.Component = function Component() {
  var _useTranslation = (0, _reactI18next.useTranslation)(),
    t = _useTranslation.t;
  var var1 = \`templateElement你好\`;
  return <div data-i18n=\\"[\\\\\\"key_\\\\u4F60\\\\u597D\\\\\\"]\\">{t(\\"key_你好\\")}</div>;
};"
`;
